src/main/java/org/sefglobal/scholarx/repository/ProfileRepository.java

```diff
@@ -2,13 +2,18 @@
 
 import org.sefglobal.scholarx.model.Profile;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
 import org.springframework.stereotype.Repository;
 
+import java.util.Map;
 import java.util.Optional;
 
 @Repository
 public interface ProfileRepository extends JpaRepository<Profile, Long> {
     Optional<Profile> findByUid(String uid);
+    Profile findByEmail(String email);
     Boolean existsByUid(String uid);
     Boolean existsByEmail(String email);
+    Profile processUserRegistration(Map<String, Object> attributes, OidcIdToken idToken, OidcUserInfo userInfo);
```

Comment:
I think we don't need this here. It is already implemented in `src/main/java/org/sefglobal/scholarx/service/ProfileService.java`. (Need to update the input params)
```suggestion
```